website/cue/reference/components/transforms.cue

```diff
@@ -18,5 +18,9 @@ components: transforms: [Name=string]: {
 		utilization:                          components.sources.internal_metrics.output.metrics.utilization
 		component_sent_events_total:          components.sources.internal_metrics.output.metrics.component_sent_events_total
 		component_sent_event_bytes_total:     components.sources.internal_metrics.output.metrics.component_sent_event_bytes_total
+		buffer_received_events_total:         components.sources.internal_metrics.output.metrics.buffer_received_events_total
+		buffer_received_bytes_total:          components.sources.internal_metrics.output.metrics.buffer_received_bytes_total
+		buffer_sent_events_total:             components.sources.internal_metrics.output.metrics.buffer_sent_events_total
+		buffer_sent_bytes_total:              components.sources.internal_metrics.output.metrics.buffer_sent_bytes_total
```

Comment:
I think you meant to add these to `sinks.cue`?

Reply:
Ah, I also added to `transforms.cue` because [transforms use buffers internally](https://github.com/vectordotdev/vector/blob/master/src/topology/builder.rs#L169-#L173). Happy to remove as it seems an opaque detail to users, but with the way the instrumentation currently works, buffer metrics will appear tagged with transform context if transforms are used.

Reply:
Ah, yeah, I forgot we re-use buffers there too. I think we should probably hide that implementation detail by not having those buffers report metrics. This could be a flag that is passed to the `vector_core::buffers::Variant::Memory`.

I'm actually curious what the motivation was to have a memory buffer there. Maybe @lukesteensen knows?